pkg/license/license.go

```diff
@@ -0,0 +1,386 @@
+			regexp.MustCompile(`(?i)^` + copyingRegex.String() + `$`):                                          0.90, // COPYING
+			regexp.MustCompile(`(?i)^` + copyingRegex.String() + preferredExtRegex.String() + `$`):             0.85, // COPYING.md
+			regexp.MustCompile(`(?i)^` + licenseRegex.String() + anyExtRegex.String() + `$`):                   0.80, // LICENSE.textile
+			regexp.MustCompile(`(?i)^` + copyingRegex.String() + anyExtRegex.String() + `$`):                   0.75, // COPYING.textile
+			regexp.MustCompile(`(?i)^` + licenseRegex.String() + `[-_][^.]*` + anyExtRegex.String() + `?$`):    0.70, // LICENSE-MIT
+			regexp.MustCompile(`(?i)^` + copyingRegex.String() + `[-_][^.]*` + anyExtRegex.String() + `?$`):    0.65, // COPYING-MIT
+			regexp.MustCompile(`(?i)^\w+[-_]` + licenseRegex.String() + `[^.]*` + anyExtRegex.String() + `?$`): 0.60, // MIT-LICENSE-MIT
+			regexp.MustCompile(`(?i)^\w+[-_]` + copyingRegex.String() + `[^.]*` + anyExtRegex.String() + `?$`): 0.55, // MIT-COPYING
+			regexp.MustCompile(`(?i)^` + oflRegex.String() + preferredExtRegex.String()):                       0.50, // OFL.md
+			regexp.MustCompile(`(?i)^` + oflRegex.String() + anyExtRegex.String()):                             0.45, // OFL.textile
+			regexp.MustCompile(`(?i)^` + oflRegex.String() + `$`):                                              0.40, // OFL
+			regexp.MustCompile(`(?i)^` + patentsRegex.String() + `$`):                                          0.35, // PATENTS
+			regexp.MustCompile(`(?i)^` + patentsRegex.String() + anyExtRegex.String() + `$`):                   0.30, // PATENTS.txt
+		}
+	)
+
+	var licenseFiles []LicenseFile
+	var ignore bool
+	err := fs.WalkDir(fsys, ".", func(filePath string, info fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if info.IsDir() {
+			return nil
+		}
+		// Let's ignore all files in the melange-out/ directory, as it's not part of the source
+		if strings.Contains(filePath, "melange-out") {
+			return nil
+		}
+
+		// Check if the file matches any of the license-related regex patterns
+		for regex, weight := range filenameRegexes {
+			if regex.MatchString(info.Name()) {
+				// licensee does this check as part of the regex, but in go we don't have
+				// the same regex capabilities
+				for _, ext := range ignoredExt {
+					if ignore = strings.HasSuffix(info.Name(), ext); ignore {
+						break
+					}
+				}
+				if ignore {
+					continue
+				}
+
+				// Licenses in the top level directory have a higher weight so that they
+				// always appear first
+				if filepath.Dir(filePath) == "." {
+					weight += 0.5
+				}
+				licenseFiles = append(licenseFiles, LicenseFile{
+					Name:   info.Name(),
+					Path:   filePath,
+					Weight: weight,
+				})
+				break
+			}
+		}
+
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// Sort the license files by their 'score' (weight)
```

Comment:
Can weights ever be tied? If so, should we have some tiebreaker field? Alternatively use `sort.SliceStable`?